projects/vdk-core/src/vdk/internal/core/config.py

```diff
@@ -124,6 +124,19 @@ def is_sensitive(self, key: ConfigKey) -> bool | None:
         """
         return self.__config_key_to_sensitive.get(key)
 
+    def is_default(self, key: ConfigKey) -> bool:
+        """
+        Return True if the configuration value associated with a given key
+        is the same as the default value
+
+        :param key: the configuration key (e.g db_host, service_uri, etc.)
+        :return: the value corresponding to the configuration key
+        """
+        key = _normalize_config_key(key)
+        default_value = self.__config_key_to_default_value.get(key)
+        value = self.__config_key_to_value.get(key, default_value)
+        return value == default_value
```

Comment:
That would not work . If the default for some option is is "XXX" and the user also sets it to "XXX". 
That would erroneously return that the we are using the default . And it should not.